arch/arc/core/offsets/offsets.c

```diff
@@ -115,6 +115,17 @@ GEN_OFFSET_SYM(_callee_saved_stack_t, dpfp2l);
 GEN_OFFSET_SYM(_callee_saved_stack_t, dpfp1h);
 GEN_OFFSET_SYM(_callee_saved_stack_t, dpfp1l);
 #endif
+#endif
+#ifdef CONFIG_DSP
+GEN_OFFSET_SYM(_callee_saved_stack_t, dsp_ctrl);
```

Comment:
Does it make sense to save the control registers (DSP_CTRL and DSP_FFT_CTRL)?
I guess it depends on the application, but the rounding/saturation etc. settings may be a global choice, not per task.

Reply:
I'm not quite familiar with DSP. But I think If two threads are running two different DSP applications, DSP_CTRL is quite likely different. And Jacco in MLI team told me that, MLI sets the dsp mode on every call, make sure that at least accumulator registers and dsp mode are saved and restored.

Reply:
ok, if also DSP mode is used dynamically and not statically configured, then it should be saved. Maybe ask @JaccovG to review this PR as well?